### PR TITLE
Add code of conduct contributing guide and docs

### PR DESCRIPTION
**Pull request by finsberg**
Adding code of conduct, contributing guide and docs.
Also adding add a placeholder for a demo of real function spaces (@jorgensd should probably write a more comprehensive example).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+pre-commit run --all
+```
+To learn more about pre-commit you can check out https://pre-commit.com
+
+## Test suite
+For every new feature of bugfix you should also make sure to not lower the code coverage for the test suite. This means that if you for example add a new function then you should also make sure that the function is properly tested (at a minimum it should be covered by the test suite).
+
+To run the test suite, please install the package with the optional dependencies `test`, i.e
+```
+python3 -m pip install -e ".[test]"
```
Comment (jorgensd) [inline CONTRIBUTING.md:56]:
```suggestion
python3 -m pip install -e ".[all]"
```
Hvis vi vil sørge for at alle tester med varierende dependencies skal støttes, eventuelt legge inn `pytest.mark.skip`.
